src/color_picker.coffee

```diff
@@ -216,10 +216,14 @@ Ember.Widgets.ColorPicker = Ember.Component.extend
       selectedColor in row.invoke 'toLowerCase'
     @set 'customColor', @get('selectedColor')
 
-  setCustomColor: Ember.observer ->
-    if @get('isCustomColorValid') is true
-      @set 'selectedColor', expandHexColor(@get('customColor'))
-  , 'customColor', 'isCustomColorValid'
+  setCustomAsSelected: Ember.observer ->
+    selectedColor = @get 'selectedColor'
+    colorRows = @get 'colorRows'
+    for row in colorRows
+      if selectedColor in row
```

Comment:
Is selected color guaranteed to be a six digit hex code? If not, this might not always work; for instance `'#999' !== '#999999'` but those colors are equivalent.